integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/QueryStringPredicateSpecificsIT.java

```diff
@@ -226,6 +233,204 @@ void enablePositionIncrements_default() {
 				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_2 );
 	}
 
+	@ParameterizedTest
+	@MethodSource
+	void queryStringSyntax(String field, String value1, String value2, String noMatch, String unParsableValue) {
+		StubMappingScope scope = indexForSyntaxParsingCheck.createScope();
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1" );
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 + "^10" ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1" );
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 + " || " + value2 )
+								.defaultOperator( BooleanOperator.AND ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2" );
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 + " OR " + value2 )
+								.defaultOperator( BooleanOperator.AND ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2" );
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 + " && " + value2 ) )
+						.fetchAllHits()
+		).isEmpty();
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field ).matching( value1 + " AND " + value2 ) )
+						.fetchAllHits()
+		).isEmpty();
+
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field )
+								.matching( String.format( Locale.ROOT, "(%s OR %s) AND !%s", value1, value2, value1 ) ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "2" );
+
+		// ranges:
+		assertThatHits(
+				scope.query()
+						.where( f -> f.queryString().field( field )
+								.matching( String.format( Locale.ROOT, "[%s TO %s]", value1, value2 ) ) )
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2" );
```

Comment:
Awesome!

BTW, I know that the fact non-string fields were not supported is not really explicit in the documentation, but now they _are_ supported you might want to explicitly mention it in the documentation of `simpleQueryString`/`queryString`? E.g. a subsection section titled "Expected format of field values" would totally make sense.

Good luck for the Elasticsearch part though ;D Maybe you should link to the table in `mapping-directfieldmapping-supported-types` for Lucene, but to the table in https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#backend-elasticsearch-field-types-available for Elasticsearch/OpenSearch.

It would probably also make sense to mention somewhere that some syntax constructs in the `queryString` predicate are only available on text fields -- e.g. regexp, phrase, etc.